### PR TITLE
Feature/#45 hr add students

### PR DESCRIPTION
**Pull request by RafalG1991**
Added guarded endpoints for hr to add and remove students from the recruitment list. I have modified the user entity - added reservation date, student status and made relations STUDENT <-> HR. I have updated the findAll students endpoint to return data about the hr that added student to list and to return the data about student status - if it's available to add or it it already added by other hr

### CODE REVIEW

**Review by piotr07 — APPROVED** [review]

File: src/student/student.service.ts [hunk 1]
```diff
@@ -51,6 +55,10 @@ export class StudentService {
         courseEngagement: true,
         projectDegree: true,
         teamProjectDegree: true,
+        hr: {
+          email: true,
+        },
+        status: true,
       },
```
Comment (piotr07) [inline src/student/student.service.ts:63]: I can't see 'reservedTo'

File: types/user/user.ts [hunk 2]
```diff
@@ -57,6 +58,10 @@ export interface User {
   courseEngagement: Score;
   projectDegree: Score;
   teamProjectDegree: Score;
+  hr: {
+    email: string;
+  };
+  status: Status;
```
Comment (piotr07) [inline types/user/user.ts:64]: I can't see 'reservedTo'

File: src/hr/hr.controller.ts [hunk 3]
```diff
@@ -17,4 +26,18 @@ export class HrController {
...
+  addStudent(@UserObj() hr: User, @Body() student: { email: string }) {
+    return this.hrService.addStudent(student.email, hr);
+  }
+
+  @Roles(Role.HR)
+  @UseGuards(AuthGuard('jwt'), RoleGuard)
+  @Delete()
+  removeStudent(@UserObj() hr: User, @Body() student: { email: string }) {
+    return this.hrService.removeStudent(student.email, hr);
+  }
```
Comment (piotr07) [inline src/hr/hr.controller.ts]: Missed endpoint for user(student) with status hired

**Review by piotr07 — COMMENTED** [review]
I can't see 'reservedTo'

File: src/student/student.service.ts (hunk 1 above)
Comment (RafalG1991) [reply src/student/student.service.ts:63]: reservedTo has been added to findAll students method

File: types/user/user.ts (hunk 2 above)
Comment (RafalG1991) [reply types/user/user.ts:64]: reservedTo has been added to User interface

File: src/hr/hr.controller.ts (hunk 3 above)
Comment (RafalG1991) [reply src/hr/hr.controller.ts]: Added endpoint for hiring students with email information and own email template. Changed the http method in removeStudent from DELETE to PATCH. Added validation in removeStudent when hr tries to remove already hired student from the list

**Pull request merged**